astropy/utils/masked/core.py

```diff
@@ -816,18 +826,35 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
             # If relevant, also mask output elements for which where was masked.
             if where_mask is not None:
                 mask |= where_mask
+            if out_mask is not None:
+                # Check for any additional explicitly given outputs.
+                for m in out_masks[1:]:
+                    if m is not None and m is not out_mask:
+                        m[...] = mask
```

Comment:
just curious, why not use `np.copyto` here ? is this significantly faster ?

Reply:
Ah, I remember now, `np.copyto` does broadcasting, so when one also has a `where` argument, one has no choice.